service/src/main/java/greencity/security/service/AuthorityServiceImpl.java

```diff
@@ -48,6 +52,18 @@ public void updateEmployeesAuthorities(UserEmployeeAuthorityDto dto, String emai
         saveNewAuthorities(dto, employee);
     }
 
+    @Override
+    public void updateAuthorities(UpdateEmployeeAuthoritiesDto dto) {
+        User employee = userRepo.findByEmail(dto.getEmail()).orElseThrow(
+            () -> new UsernameNotFoundException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL + dto.getEmail()));
+        deleteOldAuthorities(employee);
+        List<String> positionNames = dto.getPositions().stream()
+            .map(PositionDto::getName).collect(Collectors.toList());
+        List<Authority> list = authorityRepo.findAuthoritiesByPositions(positionNames);
```

Comment:
can different positions have the same authority?
If yes please use set to omit authorities duplication

Reply:
I am using distinct in query, I suppose it is enough.